modules/custom/az_migration/src/Plugin/migrate/source/AZNode.php

```diff
@@ -44,4 +60,28 @@ public function prepareRow(Row $row) {
     return parent::prepareRow($row);
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function query() {
+    $query = parent::query();
+    /*
+     * Filter date from migration script can be a
+     * string date or a UNIX timestamp so get
+     * it into correct format
+     */
+    if (array_key_exists("filter_date", $this->configuration)) {
+      if (is_int($this->configuration['filter_date'])) {
+        $date_cutoff = $this->configuration['filter_date'];
+      }
+      else {
+        $date_cutoff = (new \DateTime($this->configuration['filter_date']))->getTimestamp();
+      }
+
+      $query->condition('n.changed', $date_cutoff, '>');
+    }
+
+    return $query;
+  }
+
```

Comment:
/**
   * {@inheritdoc}
   */
  public function is_timestamp($timestamp)
{
   return ((string) (int) $timestamp === $timestamp)
   && ($timestamp <= PHP_INT_MAX)
   && ($timestamp >= ~PHP_INT_MAX)
   && (!strtotime($timestamp));
}

Reply:
What do you think about this approach @rmarsh1ua

Reply:
It blows my mind that this isn't something built into PHP as a general validation function. Looks good, but should it be a private function?

Reply:
Yeah, I went down a rabbit hole looking for a built-in too, ha. Probably should be private